app/liquid_tags/codesharer_tag.rb

```diff
@@ -0,0 +1,54 @@
+class CodesharerTag < LiquidTagBase
+  PARTIAL = "liquids/codesharer".freeze
+  PARAM_REGEXP = /\w|(#|=)/.freeze
+
+  def initialize(_tag_name, link, _parse_context)
+    super
+    stripped_link = ActionController::Base.helpers.strip_tags(link)
+    the_link = stripped_link.split(" ").first
+    @embedded_url = CodesharerTag.embedded_url(the_link)
+  end
+
+  def render(_context)
+    ApplicationController.render(
+      partial: PARTIAL,
+      locals: {
+        url: @embedded_url
+      },
+    )
+  end
+
+  def self.embedded_url(link)
+    "https://codesharer.netlify.app/#{URI.encode_www_form(parse_link(link))}"
+  end
+
+  def self.parse_link(link)
+    begin
+      url = URI(link)
+    rescue StandardError
+      raise_error
+    end
+    hostname_ok = url.hostname == "codesharer.netlify.app"
+    raise_error unless hostname_ok
```

Comment:
I don't think the extra variable adds much here.

```ruby
raise_error unless url.hostname == CODESHARER_URL
pare_params(url)
```

Reply:
Thanks so much for the review! 

Code Sharer is my repo. I just want to be able to share nice code snippets in my dev.to posts, not to advertise the project. 

RE: site reliability - I care about SRE a lot. I don't expect that to be an issue. The CI pipeline keeps errors from arriving in production.

Reply:
Thanks for the context @genderev! I hope you enjoy learning Ruby and contributing to the Forem codebase 😃